src/MongoDB.Driver.Core/Core/Configuration/ClusterSettings.cs

```diff
@@ -35,7 +35,11 @@ public class ClusterSettings
         #endregion
 
         // fields
+#pragma warning disable 618
         private readonly ClusterConnectionMode _connectionMode;
+#pragma warning restore 618
+//        private readonly bool _connectonModeWasSet;
```

Comment:
this logic should be restored. In the current implementation, we will always see thrown exception if DirectConnection has been set. I will work on it tomorrow